app/plan.go

```diff
@@ -51,7 +53,32 @@ func (s *planService) FindByName(name string) (*appTypes.Plan, error) {
 
 // DefaultPlan implements DefaultPlan method of PlanService interface
 func (s *planService) DefaultPlan() (*appTypes.Plan, error) {
-	return s.storage.FindDefault()
+	plan, err := s.storage.FindDefault()
+	if err != nil && err == appTypes.ErrPlanDefaultNotFound {
+		plans, err := s.storage.FindAll()
+		if err != nil {
+			log.Errorf("Error listing Plans: %s", err)
+			return nil, appTypes.ErrPlanDefaultNotFound
+		}
+		if len(plans) == 0 {
+			configMemory, _ := config.GetInt("docker:memory")
+			configSwap, _ := config.GetInt("docker:swap")
+			df := appTypes.Plan{
+				Name:     "autogenerated",
+				Memory:   int64(configMemory) * 1024 * 1024,
+				Swap:     int64(configSwap-configMemory) * 1024 * 1024,
+				CpuShare: 100,
+				Default:  true,
+			}
+			err := s.storage.Insert(df)
+			if err != nil {
+				log.Errorf("Error inserting autogenerated Default Plan: %s", err)
+				return nil, appTypes.ErrPlanDefaultNotFound
```

Comment:
I also think the original error should be returned here

Reply:
But return an error about insert plan when the request is about create app don't will confuse the end user?